tests/functional/retry/test_retry.py

```diff
@@ -291,3 +291,36 @@ def test_retry(self, project):
         run_dbt(["run", "--project-dir", "proj_location_1"], expect_pass=False)
         move(proj_location_1, proj_location_2)
         run_dbt(["retry", "--project-dir", "proj_location_2"], expect_pass=False)
+
+
+class TestRetryVars:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "sample_model.sql": "select {{ var('myvar_a') + var('myvar_b') }} as mycol",
+        }
+
+    def test_retry(self, project):
+        run_dbt(["run", "--vars", '{"myvar_a": "12", "myvar_b": "3 4"}'], expect_pass=False)
+        move("target", "state")
+        results = run_dbt(["retry", "--state", "state", "--vars", '{"myvar_b": "34"}'])
```

Comment:
Should we try to combine the `vars` user supported across retries? Or should we consider `vars` to be one unit and have user also supply `myvar_a` also during retry?